app/scripts/models/auth_brokers/web-channel.js

```diff
@@ -96,20 +96,78 @@ define([
                 this, account, { closeWindow: true });
     },
 
+    beforeSignUpConfirmationPoll: function (account) {
+      // If the relier wants keys, the signup verification tab will need
+      // to be able to fetch them in order to complete the flow.
+      // Send them as part of the oauth session data.
+      if (this.relier.wantsKeys()) {
+        this.session.set('oauth', _.extend({}, this.session.oauth, {
+          keyFetchToken: account.get('keyFetchToken'),
+          unwrapBKey: account.get('unwrapBKey')
+        }));
+      }
+    },
+
+    afterSignUpConfirmationPoll: function (account) {
+      // The original tab can finish the OAuth flow if it is still open,
+      // but not if the verification tab has already finished it.
+      this.session.reload();
```

Comment:
Can this and the next line be combined (in all 4 places) into a descriptively named helper function that reduces the need for the comment, something like `if (this.canFinishOAuthFlow()) {` or `if (! this.hasOAuthFlowFinished()) {`